### ISSUE TITLE
 how to use the \"event-date\" sub-field to group events with the same value (date) 

### ISSUE THREAD

**Issue by alvaromartins** [Mediatejo/customfields#1]
1) I have a custom post type \"events\" organized into 2 categories: \"active\" or \"archive\" 
2) each event has several occurrences and to manage this I have created a repeater group field \"event-shows\" with the following sub-fields: \"event-date\", \"event-time\", \"event-venue\" (this last is relational type) 
3) I want to list the events of the \"active\" category grouped by day (event-date) and sorted by hour (event-time), and the same event may appear on different days at different times (as result of multiple occurrences of the repeater field). 

example: 

13.7.2018 
11:00 — EVENT A, VENUE 1 
12:00 — EVENT B, VENUE 1 
13:00 — EVENT C, VENUE 2 

14.7.2018 
10:00 — EVENT A, VENUE 2 
12:00 — EVENT A, VENUE 1 
12:00 — EVENT D, VENUE 1 

etc 

basically, I don\'t know how to use the \"event-date\" sub-field to group events with the same value (date)

**Comment by alvaromartins** [comment]
Jul 8, 9:43pm
Hi James Thanks for you reply. For another content I was able to generate a list of \"venues\" grouped by a custom field \"city\" using the following: 

check programa.php.txt bellow.


I would suggest that you take a look at the array_multisort() function to that will make it possible to sort a repeater using one the sub fields in this case the date field: https://www.advancedcustomfields.com/resources/how-to-sorting-a-repeater-field/

I hope this info helps.

**Comment by alvaromartins** [comment]
nathanaelphilip wrote:

does this help: https://support.advancedcustomfields.com/forums/topic/order-by-subfield-values/

seems like you can do subnested orderbys now: https://make.wordpress.org/core/2015/03/30/query-improvements-in-wp-4-2-orderby-and-meta_query/

(i didn’t know about this until now – very exciting!)

Phil
				
Post Link: https://support.advancedcustomfields.com/forums/topic/list-posts-grouped-by-sub-field-of-repeater/#post-61310

**Comment by alvaromartins** [comment]
[programa.php.txt](https://github.com/Mediatejo/customfields/files/2185120/programa.php.txt)

	<?php
		while ( have_posts() ) :
			the_post(); ?>	
			
			<h1><?php the_title() ?></h1>
			<div><?php the_content() ?></div>
			
			<?php
			$args = array( 
				'posts_per_page' => -1,
				'post_type' => 'eventos',
				'status' => 'programa',
				'orderby'    => array(
					'title' => 'ASC')
				);
			$loop = new WP_Query( $args );
			while ( $loop->have_posts() ) : $loop->the_post(); ?>
				<hr>
				<?php the_field('evento-viagem'); ?> / <?php the_field('evento-audiencia'); ?><br>
				<a href="<?php the_permalink(); ?>"><?php the_title(); ?></a><br>
				<?php if( get_field('evento-subtitulo') ): ?>
					<?php the_field('evento-subtitulo'); ?><br>
				<?php endif; ?>
				<?php the_field('evento-caminho'); ?> <?php the_field('evento-ano'); ?><br>
					
					<?php if( have_rows('evento-paragens') ): ?>
						<ul>
						<?php while( have_rows('evento-paragens') ): the_row(); ?>
							<li>
								<?php the_sub_field('evento-data'); ?> — <?php the_sub_field('evento-hora'); ?>

								<?php 
									$posts = get_sub_field('evento-local');
									if( $posts ): ?>
										<?php foreach( $posts as $p ): // variable must NOT be called $post (IMPORTANT) ?>
												<?php echo get_the_title( $p->ID ); ?> <?php the_field('local-cidade', $p->ID); ?>
										<?php endforeach; ?>
								<?php endif; ?>
								
							</li>
						<?php endwhile; ?>
						</ul>
					<?php endif; ?>
					
			
			<?php endwhile; ?>
			
		<?php
		endwhile; // End of the loop.
		?>

**Comment by alvaromartins** [comment]
https://caminhos.mediotejo.pt/alvaro-1/ 
No WP é o link Locais no painel de administraçao

https://caminhos.mediotejo.pt/alvaro
No WP é o link Evento no painel de administraçao

Sao ambos uma pagina do tipo post_type com campos customizaveis. 
Se criares um novo evento vais ver os campos e subcampos.
A ideia é criar esse sort pela data que é um subcampo do evento.
